logstash-core/src/main/java/org/logstash/RubyUtil.java

```diff
@@ -63,6 +80,20 @@
         RUBY_EVENT_CLASS.setConstant("VERSION_ONE", RUBY.newString(Event.VERSION_ONE));
         RUBY_EVENT_CLASS.defineAnnotatedMethods(JrubyEventExtLibrary.RubyEvent.class);
         RUBY_EVENT_CLASS.defineAnnotatedConstants(JrubyEventExtLibrary.RubyEvent.class);
+        RUBY_ACKED_BATCH_CLASS = setupLogstashClass("AckedBatch", new ObjectAllocator() {
+            @Override
+            public IRubyObject allocate(final Ruby runtime, final RubyClass rubyClass) {
+                return new JrubyAckedBatchExtLibrary.RubyAckedBatch(runtime, rubyClass);
+            }
+        }, JrubyAckedBatchExtLibrary.RubyAckedBatch.class);
+        setupLogstashClass(
+            "AckedQueue", JrubyAckedQueueExtLibrary.RubyAckedQueue::new,
+            JrubyAckedQueueExtLibrary.RubyAckedQueue.class
+        );
+        setupLogstashClass(
+            "AckedMemoryQueue", JrubyAckedQueueMemoryExtLibrary.RubyAckedMemoryQueue::new,
+            JrubyAckedQueueMemoryExtLibrary.RubyAckedMemoryQueue.class
+        );
```

Comment:
+1 to getting these guys together... however we are starting to hit the limit of reasonable static initializer. Any thoughts on increasing the readability here ? (I like what did with the setupLogstashClass)

Reply:
@jakelandis I think this will clear up as soon as this logic is not `static` anymore and simply part of a normal constructor chain. I'll do my best to prepare a PR that gets us away from the Ruby global constant soon (this is fairly important for a stable/maintainable Java entry point).

For now, I'm not so sure I want to dry this up more (or spread that logic out more), just to keep the effort for moving away from `static` and towards a Java entry-point lower (a bit of a weak argument I know, but I'd rather not start abstracting things here, when I don't know what hacky challenges lie ahead :D if that makes any sense).

Reply:
> as this logic is not static anymore and simply part of a normal constructor chain
🥇